gateway/ln-gateway/src/lib.rs

```diff
@@ -251,19 +252,36 @@ impl Gateway {
         Ok(())
     }
 
-    async fn handle_get_info(&self, _payload: InfoPayload) -> Result<GatewayInfo> {
-        let federations = self
+    async fn handle_get_info(&self, payload: InfoPayload) -> Result<GatewayInfo> {
+        let federations: Vec<FederationInfo> = self
             .actors
             .lock()
             .await
             .iter()
             .map(|(_, actor)| actor.get_info().expect("Failed to get actor info"))
             .collect();
 
-        Ok(GatewayInfo {
-            federations,
-            version_hash: env!("CODE_VERSION").to_string(),
-        })
+        match payload.federation_id {
+            Some(id) => {
+                // TODO: (@oleonardolima) Improve the filter by federation_id
+                let filtered_federations = federations
+                    .iter()
+                    .filter(|federation_info| federation_info.federation_id == id)
+                    .cloned()
+                    .collect::<Vec<_>>();
+                // let filtered_federations = federations
+                //     .drain_filter(|federation_info| federation_info.federation_id == id)
+                //     .collect::<Vec<_>>();
+                Ok(GatewayInfo {
+                    federations: filtered_federations,
+                    version_hash: env!("CODE_VERSION").to_string(),
+                })
+            }
+            None => Ok(GatewayInfo {
```

Comment:
You could probably take the `Ok(GatewayInfo( { ... , version_hash: env!("CODE_VERSION").to_string(), })}` outside of the `match` to make it more compact, then use `payload.federation_id.map(|id| ... ).unwrap_or(federations)` to get rid of `match` altogether. IMO that's nicer, but one might argue.

Reply:
I updated it to filter the actor beforehand and call another method, so I went back for a match clause.

Do you think that it could be refactored somehow ?